lib/terminal-table/style.rb

```diff
@@ -62,6 +62,15 @@ def defaults= options
           @@defaults = defaults.merge(options)
         end
       end
+
+      def on_change attr
```

Comment:
@nateberkopec This helps enabling binding change on style attributes.

Since I change the way doing recalculate the width and increases its computational complexity, I don't want to recalculate the width each time, instead, recalculate only something has changed and something is asking for the column widths.

For other actions such as adding a row can be done easily within `table.rb`. However for actions taken in `style.rb` (`width` changes in this case), without affect current code, the best way I can think of is to add the `on_change` method what works like Event API in JavaScript.

Actually, more attributes should be added (such as padding and border?) but in a new PR since not related to this issue (fix current specs).